data/exploits/psnuffle/ftp.rb

```diff
@@ -40,7 +40,15 @@ def parse(pkt)
 
       when :login_fail
         if(s[:user] and s[:pass])
-          report_auth_info(s.merge({:active => false}))
+          report_cred(
+            :ip  => s[:host],
+            :port => 21,
```

Comment:
Why do we need to hard code 21 here, is it not just `s[:port]` ? 👀

Reply:
```
def parse(pkt)
    # We want to return immediatly if we do not have a packet which is handled by us
    return unless pkt.is_tcp?
    return if (pkt.tcp_sport != 21 and pkt.tcp_dport != 21)
    s = find_session((pkt.tcp_sport == 21) ? get_session_src(pkt) : get_session_dst(pkt))
    ...
```

The function only analyses packets where the target's port 21 is involved(hard coded by the writer?). So `s[:port]` is always gonna be 21. But yes, cleaner to use s[:port] instead. I'll have to do the same for the other modules too.